### PR TITLE
executor: fix DATA RACE in TestInfoSchemaClusterTable

### LINKED ISSUE #33747: [FAIL]  executor TestInfoSchemaClusterTable
Data race test case tracked in https://github.com/pingcap/tidb/issues/33745

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #33747 

Problem Summary:

### What is changed and how it works?

`make ut X='run executor TestInfoSchemaClusterTable --race'`  to reproduce it.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
it is related to #32747. It is easy for ```suite.Run``` to make data race.

**Comment by tiancaiamao** [comment]
> it is related to #32747. It is easy for `suite.Run` to make data race.

We can just initialize the testify package to avoid this race.

**Comment by tiancaiamao** [comment]
close #32747

File: executor/infoschema_cluster_table_test.go [hunk 1]
```diff
@@ -82,8 +82,9 @@ func (s *infosSchemaClusterTableSuite) setUpRPCService(addr string) (*grpc.Serve
 	srv := server.NewRPCServer(config.GetGlobalConfig(), s.dom, sm)
 	port := lis.Addr().(*net.TCPAddr).Port
 	addr = fmt.Sprintf("127.0.0.1:%d", port)
+	assert := s.Require()
 	go func() {
-		s.Require().NoError(srv.Serve(lis))
+		assert.NoError(srv.Serve(lis))
```
Comment (tisonkun) [inline executor/infoschema_cluster_table_test.go:87]:
```suggestion
		require.NoError(srv.Serve(lis))
```

**Comment by tisonkun** [comment]
Resolved in #34756.

**Pull request closed without merging**